### ISSUE TITLE
A wrong module is instantiated when several modules of the same plugin are present in the system

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#3292]
**Describe the bug**
If a plugin is shipped in three different forms - VST, VST3 and AU - and all three are installed in the current system, only the AU module can be accessed no matter which menu entry you select to load a plugin. If there's two modules - VST and VST3 - present in the system, then calling a VST3 module will result in VST module being called.

**To Reproduce**
1. Install a plugin that has 3 modules: AU, VST, VST3 (e.g.Kickstart 2 by Nicky Romero & Cableguys)
2. Load some audio
3. Go to Effect menu, then select a VST/VST3 version of this plugin
4. In the opened plugin dialog window go to 'About' and see what you're actually dealing with

**Actual behavior**
It's an AU module

**Expected behavior**
A corresponding VST/VST3 module has been opened

**Screenshots**

https://user-images.githubusercontent.com/70063970/180831931-9ff3a31f-a02e-44d9-bb17-c11039388a37.mp4

Ableton handles it just fine:
<img width="513" alt="image" src="https://user-images.githubusercontent.com/70063970/180831977-ef0ec849-cb95-49f0-b092-8729daf41688.png">

**Additional information (please complete the following information):**
 - OS: MacOS/Windows
 - Version: the latest `master` build

**Additional context**

**Comment by Ryan-Miller-Audacity** [comment]
I was able to reproduce this in most recent master on MacOS 12.5 as described with the Kickstart 2 plugin as well as a few others, with #3311 it's hard to tell if this occurs in the realtime effect panel as well since you cannot current change the sorting to specify by type, but this definitely happens from the destructive menu (as described above).

**Comment by Ryan-Miller-Audacity** [comment]
While testing with the latest master build on Windows today, I actually noticed that this is also possible to happen between VST and VST3 effects as well. I downloaded the TDR Kotelnikov plugin which includes both the VST and VST3 versions of the plugin and found that Audacity was opening the VST version of the plugin when selected (from the VST3 dropdown) rather than the VST3 option (confirmed by looking at the about context menu for the plugin as well as comparing the non-GUI interfaces which had differences). 

You can see the VST vs VST3 afflication on MacOS as well if you disable Audiounits in the preferences panel.

TLDR; this doesn't seem to only effect MacOS and seems to be more generalized to selecting a plugin (from any family) that has multiple formats installed. 

@dozzzzer Can you confirm my findings I list above? If so, we should either open a new more generic issue or rework this issue to include those details.
